public/components/trace_analytics/components/traces/span_detail_table.tsx

```diff
@@ -10,13 +10,15 @@ import moment from 'moment';
 import React, { useCallback, useEffect, useMemo, useState } from 'react';
```

Comment:
I'll suggest you create a fast-follow Issue/PR for refactor (2.6).

I find it really hard to follow the flow of this component with so many internals switched between Jaeger/Dataprepper with a `mode = .. ? ` tertiary test.

It might be an educational architecture exercise, as well as grealy clarify the code, to refactor this so that either the relevant data points can be gathered and sent as props to <SpanDetailTable>
... OR make <SpanDetailTable> a wrapper and put the relevant detail into a child component.

In the latter case, the high-level jsx might be :

```
<SpanDetailTable>
  {if mode == 'jaeger'}
    <JaegerDetailTable ... >
  { else }
    <DataPrepperDetailTable>
  { end }
</SpanDetailTable>

No sure what is the more-right architecture (opportunity for a spike!), hopefully the end-result will be a far clearer to read module.

Reply:
Good suggestion - this would make it more extensible if we add more types in the future. Will note it down, thanks!

Reply:
@pjfitzgibbons I was thinking on the same lines. This way it is easier for future additions.